lib/rig_inbound_gateway_web/connection_init.ex

```diff
@@ -40,17 +42,46 @@ defmodule RigInboundGatewayWeb.ConnectionInit do
         %{auth_tokens: [{"bearer", jwt}]} -> jwt
       end
 
-    with {:ok, jwt_subs} <-
-           Subscriptions.from_token(jwt),
+    with {:ok, jwt_subs} <- Subscriptions.from_token(jwt),
          true = String.starts_with?(request.content_type, "application/json"),
-         {:ok, query_subs} <-
-           Subscriptions.from_json(request.body),
+         {:ok, query_subs} <- Subscriptions.from_json(request.body),
          subscriptions = Enum.uniq(jwt_subs ++ query_subs),
          :ok <- SubscriptionAuthZ.check_authorization(request) do
+
+      {:ok, vconnection_pid} = request.connection_token
+      |> Codec.deserialize
+      |> case do
+        {:error, _} ->
+          # We're going to accept the connection
+          # and let VConnection handle subscription refresh and heartbeat
+          VConnection.start(self(), subscriptions, heartbeat_interval_ms, subscription_refresh_interval_ms)
+        {:ok, pid} ->
+          # We're going to try to reconnect
+          if Process.alive?(pid) do
+            # Try a reconnect
+            vpid = GenServer.call(pid, :reconnect)
+
+            if subscriptions != [] do
+              # If the client reconnects with new subscriptions, replace the existing subscriptions
+              send pid, {:set_subscriptions, subscriptions}
+            else
+              # Re-register subscriptions
+              send pid, :set_subscriptions
```

Comment:
what if I wanted to clear out my subscriptions? as I said, the newer set should always replace the existing subscriptions.

Reply:
Well...if you wanted to clear out your subscriptions, wouldn't you send an empty subscription  set? If nothing is sent, we just take the old subscriptions. That seems like intended behavior to me. If one **really** wants to clear out their subscriptions, sending an empty subscription set seems like a reasonable way to do so.

Reply:
isn't `if subscriptions != []` exactly about the empty subscription set?

Reply:
You're right. I'd need a way of checking whether `subscriptions` is set.

Reply:
Ok. Fixed this.